source/_components/calendar.todoist.markdown

```diff
@@ -0,0 +1,136 @@
+
+```yaml
+# Example configuration.yaml entry
+calendar:
+  - platform: todoist
+    token: !secret todoist_token
+    custom_projects:
+      - name: 'All Projects'
+      - name: 'Due Today'
+        due_date_days: 0
+      - name: 'Due This Week'
+        due_date_days: 7
+      - name: 'Math Homework'
+        labels:
+          - Homework
+        include_projects:
+          - Mathematical Structures II
+          - Calculus II
+```
+
+(See [here](https://home-assistant.io/docs/configuration/secrets/) for more details about what that `!secret` does -- it's not exclusive to Todoist, and can help keep your API keys and passwords a little safer!)
+
+As you can see, there are 4 custom projects here:
+
+- A project containing *all* of the tasks on this account.
+
+- A project containing *all* the tasks on this account that are due today.
+
+- A project containing *all* the tasks on this account due within the next week.
+
+- A projct containing everything with the label "Homework", taking only 2 projects into account.
+
+You can mix-and-match these attributes to create all sorts of custom projects. You can even use [IFTTT](https://ifttt.com/todoist) to create a task with a certain label, then have Home Assistant do some kind of automation when a task with that label comes due.
+
+Home Assistant does its best to determine what task in each project is "most" important, and it's that task which has its state reported. You can access the other tasks you have due soon via the `all_tasks` array (see below).
+
+### {% linkable_title Sensor attributes %}
+
+ - **offset_reached**: Not used.
+
+ - **all_day**: `True` if the reported task doesn't have a due date. `False` if there is a due date set.
+
+ - **message**: The title of the "most important" task coming up in this project.
+
+ - **description**: A URL pointing to the task on the Todoist website.
+
+ - **location**: Not used.
+
+ - **start_time**: The last time the Todoist component got updated. Usually within the last 15 minutes.
+
+ - **end_time**: When the task is due.
+
+- **all_tasks**: A list of all tasks in this project, sorted from most important to least important.
+
+- **priority**: The priority Todoist reports this task as having. 1 means lowest priority, 4 means highest. Note that this is the **opposite** of how things are displayed in the Todoist app!
+
+- **task_comments**: Any comments added to this task.
+
+- **task_labels**: All labels associated with this task.
+
+- **overdue**: Whether the reported task is past its due date.
+
+- **due_today**: Whether the reported task is due today.
+
+### {% linkable_title Services %}
+
+Todoist also comes with access to a service, `todoist.new_task`. This service can be used to create a new Todoist task. You can specify labels and a project, or you can leave them blank and the task will go to your "Inbox" project.
```

Comment:
**Spelling & Grammar**:
- Missing comma after `blank` -> `or you can leave them blank, and the task`